### PR TITLE
WIP: Support for Openshift env with non-root(oracle) user

### PR DESCRIPTION
**Pull request by ashageetha**
Hi,

This PR has fix for Apache Webtier to run in Openshift env with non-root user and with port 8080. 
This sample is being used as part of Apache Webtier Setup in Openshift env as per WebLogic Kubernetes Operator Samples: "https://github.com/oracle/weblogic-kubernetes-operator/tree/master/kubernetes/samples/charts/apache-webtier".

Currently tested in below scenario:
1. docker run -d -e WEBLOGIC_HOST=<host> -e WEBLOGIC_PORT=<port> -p 8080:8080 oracle/apache:12.2.1.3
2. Openshift env as per '"https://github.com/oracle/weblogic-kubernetes-operator/tree/master/kubernetes/samples/charts/apache-webtier"  for a non-SSL usecase.

### CODE REVIEW

**Comment by Djelibeybi** [comment]
Is this still a WIP or is it ready for review?

**Comment by ashageetha** [comment]
> 
> 
> Is this still a WIP or is it ready for review?

Its still in review as the changes impact the samples used in Operator repo: https://github.com/oracle/weblogic-kubernetes-operator/tree/master/kubernetes/samples/charts/apache-webtier.
So we are planning to finalize the changes this week.

File: OracleWebLogic/samples/12213-webtier-apache/pluginWeblogic.conf [hunk 1]
```diff
@@ -0,0 +1 @@
+/u01/oracle/lib
```
Comment (doxiao) [inline OracleWebLogic/samples/12213-webtier-apache/pluginWeblogic.conf:1]: Add copyright.

File: OracleWebLogic/samples/12213-webtier-apache/Dockerfile [hunk 2]
```diff
@@ -46,16 +47,41 @@ COPY container-scripts/* /u01/oracle/container-scripts/
...
+    ldconfig && \
+    rm /tmp/$PLUGINS_PKG /tmp/WLSPlugin12.2.1.3.0-*.zip /tmp/$FMW_PKG /tmp/fmw_12213_readme* && \
+    chown oracle:oracle -R /u01 && \
+    chown oracle:oracle -R /tmp && \
+    chown oracle:oracle -R /config && \
+    chown oracle:oracle -R /run
 
 #Expose ports
-EXPOSE 80 4433
+EXPOSE 8080 4433
```
Comment (doxiao) [inline OracleWebLogic/samples/12213-webtier-apache/Dockerfile:82]: Will this cause problems for the existing customers when they upgrade to the new version?

File: OracleWebLogic/samples/12213-webtier-apache/weblogic.conf [hunk 3]
```diff
@@ -1,7 +1,7 @@
-# Copyright (c) 2018 Oracle and/or its affiliates. All rights reserved.
+# Copyright (c) 2018-2020 Oracle and/or its affiliates.
 #
-# Licensed under the Universal Permissive License v 1.0 as shown at http://oss.oracle.com/licenses/upl.
-LoadModule weblogic_module /root/lib/${MOD_WLS_PLUGIN}
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+LoadModule weblogic_module /u01/oracle/lib/mod_wl_24.so
```
Comment (doxiao) [inline OracleWebLogic/samples/12213-webtier-apache/weblogic.conf:4]: The old version uses MOD_WLS_PLUGIN env var. Is there a reason to change it to the hard-coded value?

File: OracleWebLogic/samples/12213-webtier-apache/Dockerfile [hunk 4]
```diff
@@ -46,16 +47,41 @@ COPY container-scripts/* /u01/oracle/container-scripts/
...
 # Note that adding unzip does not noticeably increase the size of the image
-RUN yum install -y unzip httpd libaio mod_ssl.x86_64 && \
+RUN yum install -y sudo unzip httpd libaio mod_ssl.x86_64
+
+RUN groupadd sudo && \
+    useradd -b /u01 -d /u01/oracle -m -G sudo -s /bin/bash oracle && \
+    touch /etc/sudoers && \
+    sed -i /etc/sudoers -re 's/^%sudo.*/%sudo ALL=(ALL:ALL) NOPASSWD: ALL/g' && \
+    sed -i /etc/sudoers -re 's/^root.*/root ALL=(ALL:ALL) NOPASSWD: ALL/g' && \
+    sed -i /etc/sudoers -re 's/^#includedir.*/## **Removed the include directive** ##"/g' && \
```
Comment (doxiao) [inline OracleWebLogic/samples/12213-webtier-apache/Dockerfile:57]: The include directive is commented out. Do we really need to remove it?

File: OracleWebLogic/samples/12213-webtier-apache/pluginWeblogic.conf (hunk 1 above)
Comment (ashageetha) [reply OracleWebLogic/samples/12213-webtier-apache/pluginWeblogic.conf:4]: Added copyright header.

File: OracleWebLogic/samples/12213-webtier-apache/Dockerfile (hunk 2 above)
Comment (ashageetha) [reply OracleWebLogic/samples/12213-webtier-apache/Dockerfile:82]: When customers upgrade to next version, for eg., say 12.2.1.4, then we plan to provide support for both 80 and 8080.

Comment (mriccell) [reply OracleWebLogic/samples/12213-webtier-apache/Dockerfile:82]: I would like to make this a generic image for 12.2.1.3, 12.2.14, 14.1.1.0  can we support both 80 and 8080 now?

**Comment by ashageetha** [comment]
> 
> 
> @ashageetha Could you reply to the two outstanding comments from Dongbo?

Sure. We are working on to support both 80 and 8080 this week. I will update you once its ready.

**Comment by Djelibeybi** [comment]
I have added WIP to the title of the PR so it doesn't get merged too early. Once you are happy with the code, please remove that and the merge will be allowed.

**Pull request closed without merging**